homeassistant/components/bme280/sensor.py

```diff
@@ -1,147 +1,158 @@
+    SENSOR_TEMP,
+    SENSOR_TYPES,
 )
 
+_LOGGER = logging.getLogger(__name__)
+
 
 async def async_setup_platform(hass, config, async_add_entities, discovery_info=None):
     """Set up the BME280 sensor."""
-
+    if discovery_info is None:
+        return
     SENSOR_TYPES[SENSOR_TEMP][1] = hass.config.units.temperature_unit
-    name = config[CONF_NAME]
-    i2c_address = config[CONF_I2C_ADDRESS]
-
-    bus = smbus.SMBus(config[CONF_I2C_BUS])
-    sensor = await hass.async_add_executor_job(
-        partial(
-            BME280,
-            bus,
-            i2c_address,
-            osrs_t=config[CONF_OVERSAMPLING_TEMP],
-            osrs_p=config[CONF_OVERSAMPLING_PRES],
-            osrs_h=config[CONF_OVERSAMPLING_HUM],
-            mode=config[CONF_OPERATION_MODE],
-            t_sb=config[CONF_T_STANDBY],
-            filter_mode=config[CONF_FILTER_MODE],
-            delta_temp=config[CONF_DELTA_TEMP],
-            logger=_LOGGER,
+    sensor_conf = discovery_info[SENSOR_DOMAIN]
+    name = sensor_conf[CONF_NAME]
+    scan_interval = sensor_conf[CONF_SCAN_INTERVAL]
+    if CONF_SPI_BUS in sensor_conf and CONF_SPI_DEV in sensor_conf:
+        spi_dev = sensor_conf[CONF_SPI_DEV]
+        spi_bus = sensor_conf[CONF_SPI_BUS]
+        _LOGGER.debug("BME280 sensor initialize at %s.%s", spi_bus, spi_dev)
+        interface = INTERFACE_SPI
+        sensor = await hass.async_add_executor_job(
+            partial(
+                BME280_spi,
+                t_mode=sensor_conf[CONF_OVERSAMPLING_TEMP],
+                p_mode=sensor_conf[CONF_OVERSAMPLING_PRES],
+                h_mode=sensor_conf[CONF_OVERSAMPLING_HUM],
+                standby=sensor_conf[CONF_T_STANDBY],
+                filter=sensor_conf[CONF_FILTER_MODE],
+                spi_bus=sensor_conf[CONF_SPI_BUS],
+                spi_dev=sensor_conf[CONF_SPI_DEV],
+            )
+        )
+        if not sensor.sample_ok:
+            _LOGGER.error("BME280 sensor not detected at %s.%s", spi_bus, spi_dev)
+            return
+        sensor_update = sensor.update
+    else:
+        i2c_address = sensor_conf[CONF_I2C_ADDRESS]
+        bus = smbus.SMBus(sensor_conf[CONF_I2C_BUS])
+        interface = INTERFACE_I2C
+        sensor = await hass.async_add_executor_job(
+            partial(
+                BME280_i2c,
+                bus,
+                i2c_address,
+                osrs_t=sensor_conf[CONF_OVERSAMPLING_TEMP],
+                osrs_p=sensor_conf[CONF_OVERSAMPLING_PRES],
+                osrs_h=sensor_conf[CONF_OVERSAMPLING_HUM],
+                mode=sensor_conf[CONF_OPERATION_MODE],
+                t_sb=sensor_conf[CONF_T_STANDBY],
+                filter_mode=sensor_conf[CONF_FILTER_MODE],
+                delta_temp=sensor_conf[CONF_DELTA_TEMP],
+            )
         )
+        if not sensor.sample_ok:
+            _LOGGER.error("BME280 sensor not detected at %s", i2c_address)
+            return
+        sensor_update = partial(sensor.update, True)
```

Comment:
What does the `True` argument mean?

Reply:
It's good that you paid attention to this. This is an optional parameter meaning first_reading. I will remove it, because the first reading occurs when the sensor is initialized.